examples/workspaces/main.tf

```diff
@@ -72,3 +58,53 @@ resource "aws_workspaces_ip_group" "main" {
     description = "Contractors"
   }
 }
+
+data "aws_region" "current" {}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+
+  filter {
+    name   = "opt-in-status"
+    values = ["opt-in-not-required"]
+  }
+}
+
+locals {
+  # Workspace instances are not supported in all AZs in some regions
+  region_workspaces_az_ids = {
+    "us-east-1" = formatlist("use1-az%d", [2, 4, 6])
+  }
+
+  workspaces_az_ids = lookup(local.region_workspaces_az_ids, data.aws_region.current.name, data.aws_availability_zones.available.zone_ids)
+}
+
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "private-a" {
+  vpc_id               = "${aws_vpc.main.id}"
+  availability_zone_id = "${local.workspaces_az_ids[0]}"
+  cidr_block           = "10.0.1.0/24"
+}
+
+resource "aws_subnet" "private-b" {
+  vpc_id               = "${aws_vpc.main.id}"
+  availability_zone_id = "${local.workspaces_az_ids[1]}"
+  cidr_block           = "10.0.2.0/24"
+}
+
+resource "aws_directory_service_directory" "example" {
+  name     = "workspaces.example.com"
+  password = "#S1ncerely"
+  size     = "Small"
+  vpc_settings {
+    vpc_id     = "${aws_vpc.main.id}"
+    subnet_ids = ["${aws_subnet.private-a.id}", "${aws_subnet.private-b.id}"]
+  }
+}
+
+data "aws_kms_key" "workspaces_default" {
+  key_id = "alias/aws/workspaces"
+}
```

Comment:
This key won't exist until it is created the first time. Should the example instead create a custom CMK?